vllm/worker/worker.py

```diff
@@ -164,12 +164,28 @@ def determine_num_available_blocks(self) -> Tuple[int, int]:
             "Error in memory profiling. This happens when the GPU memory was "
             "not properly cleaned up before initializing the vLLM instance.")
 
+        assert total_gpu_memory > peak_memory, (
+            "peak_memory = {} is more than total_gpu_memory = {}".format(
+                peak_memory, total_gpu_memory))
+
+        # In non-eager mode, we also need to add cuda graph
+        # memory to peak_memory
+        if not self.model_config.enforce_eager:
+            cuda_graph_memory = 1 * (1000 * 1024 * 1024)  # 1GB estimate
```

Comment:
this might not be accurate. it can vary according to the model size.

Reply:
yeah, the problem is that the capture happens only after the code determines gpu_blocks. I have tried this for 70B model - it works. We can also increase it to be on the conservative side (if needed).